### ISSUE TITLE
in schema, optional validator as first param causes error

### ISSUE THREAD

**Issue by smundro** [ctavan/express-validator#403]
We've found that placing `optional` as the first param for a field in a schema will lead to an error, whereas putting it at the end will result in proper functionality.

```
{
   // this works
   'field' :  { isNumeric : { errorMessage : 'must be a number' }, optional: true }
   // this causes exception below
   'field' :  { optional: true, isNumeric : { errorMessage : 'must be a number' } }
}
```

```
TypeError: Cannot set property 'message' of undefined
    at Function.middleware.withMessage.message [as withMessage] (/project/node_modules/express-validator/check/check.js:55:47)
    at validateSchema (/project/node_modules/express-validator/lib/express_validator.js:219:12)
    at IncomingMessage.req.(anonymous function) [as checkBody] (/project/node_modules/express-validator/lib/express_validator.js:384:18)
    ...
```
I couldn't find any mention of parameter ordering for schemas.  I also wonder if `optional` is in the middle it may work but no provide an accurate validation result.

**Comment by gustavohenke** [comment]
You're correct with your diagnosis; V8 has no iteration order specified for object keys, but most of the time it just iterates in the order the keys were defined.

Fortunately, this is already fixed in `master` as per 21ef4403e9fcf281f1d3cd368844d09ee573cd25. New release coming soon!

**Comment by gustavohenke** [comment]
Fix published in v4.1.1.
